atd-vzd/migrations/default/1698277990422_create_location_id_triggers/up.sql

```diff
@@ -0,0 +1,21 @@
+CREATE OR REPLACE TRIGGER update_cr3_location_on_cris_update
+BEFORE UPDATE ON cris_facts.atd_txdot_crashes
+FOR EACH ROW
+WHEN (OLD.latitude IS DISTINCT FROM NEW.latitude OR OLD.longitude IS DISTINCT FROM NEW.longitude)
+EXECUTE PROCEDURE public.update_cr3_location();
+
+CREATE OR REPLACE TRIGGER update_cr3_location_on_vz_update
+BEFORE UPDATE ON vz_facts.atd_txdot_crashes
+FOR EACH ROW
+WHEN ((OLD.latitude IS DISTINCT FROM NEW.latitude) OR (OLD.longitude IS DISTINCT FROM NEW.longitude))
+EXECUTE PROCEDURE public.update_cr3_location();
+
+CREATE OR REPLACE TRIGGER update_cr3_location_on_cris_insert
+BEFORE INSERT ON cris_facts.atd_txdot_crashes
+FOR EACH ROW
+EXECUTE PROCEDURE public.update_cr3_location();
+
+CREATE OR REPLACE TRIGGER update_cr3_location_on_vz_insert
+BEFORE INSERT ON vz_facts.atd_txdot_crashes
+FOR EACH ROW
+EXECUTE PROCEDURE public.update_cr3_location();
```

Comment:
I'm actually realizing this trigger isn't necessary because new inserts to vz_facts.atd_txdot_crashes should always have null values, correct? Have we decided how we are going to handle temp crash records in the LDM? Because maybe that would be a situation in which we would need this trigger? I'll leave this in for now depending on what y'all think but I can update the up/down migrations and delete this trigger.

Reply:
I don't think that we need it either, and I tested removing this trigger and see `NULL` in `vz_facts`. I thought that the order of the triggers would be the opposite, but using `EXPLAIN ANALYZE` shows that the trigger that copies the record from `cris_facts` to `vz_facts` happens before the trigger that populates `location_id` in `cris_facts`.

<img width="483" alt="Screenshot 2023-10-26 at 5 08 58 PM" src="https://github.com/cityofaustin/atd-vz-data/assets/37249039/df2f8bdd-d6df-48ba-a5db-962963769eb0">